build/tools/SharedCodeGenerator/VersionInfo.cs

```diff
@@ -0,0 +1,62 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Oryx.SharedCodeGenerator
+{
+    public class VersionInfo : IComparable<VersionInfo>
+    {
+        private readonly List<string> nonSemverPlatforms = new List<string> { "golang" };
+
+        public VersionInfo(string displayVersion, string platform)
+        {
+            this.DisplayVersion = displayVersion;
+            this.Platform = platform;
+            if (this.nonSemverPlatforms.Contains(platform))
+            {
+                this.Version = new Version(this.DisplayVersion);
+            }
+            else
+            {
+                this.SemanticVersion = ToSemanticVersion(this.DisplayVersion);
+            }
+        }
+
+        public string DisplayVersion { get; }
+
+        public string Platform { get; }
+
+        public SemVer.Version SemanticVersion { get; }
+
+        public Version Version { get; }
+
+        public int CompareTo(VersionInfo other) =>
+            this.nonSemverPlatforms.Contains(this.Platform)
+            ? this.Version.CompareTo(other.Version)
+            : this.SemanticVersion.CompareTo(other.SemanticVersion);
+
+        private static SemVer.Version ToSemanticVersion(string displayVersion)
+        {
+            var semanticVersionStr = displayVersion;
```

Comment:
Would you mind outlining some example scenarios for this method? I'm trying to understand a few things going on in this method, but just want to make sure I first understand its purpose.

Reply:
So I pulled this from elsewhere in the code base, but I investigated the purpose of this method and it looks like adding the loose parameter will allow certain invalid formats, such as pre-release versions without a leading hyphen e.g. 3.10.0a2. And most of the method seems to be adding the hyphen into a version number in some cases when it wasn't found. So I ended up just removing the whole thing in favor of always using the loose param for semver versions. Tested it and it works identically.

Reply:
library docs: https://github.com/adamreeve/semver.net

Reply:
Got it -- thanks for the context